lib/customer.rb

```diff
@@ -0,0 +1,34 @@
+class Customer
+
+  attr_reader :basket
+
+  def initialize
+    @basket = []
+  end
+
+  def view_menu(takeaway)
+    takeaway.menu
+  end
+
+  def add_item(takeaway, _item)
+    fail 'Sorry that item is not on the menu' if takeaway.menu.key?(_item) == false
+    @basket << _item
+  end
+
+  def place_order(takeaway)
+    puts "Send us your number to confirm the order"
+    @customer_number = "+447850122007"
```

Comment:
No need for this to be an instance variable. The `takeaway` in the next line can _only_ access it if you pass this string as a parameter to the takeaway.